spring-session/src/main/java/org/springframework/session/MapSession.java

```diff
@@ -40,7 +40,7 @@
  * @since 1.0
  * @author Rob Winch
  */
-public final class MapSession implements ExpiringSession, Serializable {
+public class MapSession implements ExpiringSession, Serializable {
```

Comment:
Ok that makes more sense then.

Would you be interested in contributing the FST serialization in a separate PR and making the class non-final in that PR rather than this one? If so, can you create an issue to track that? Thanks for considering this!

NOTE: You might hold off on sending that PR till we get some of the other comments sorted out so you don't have dual maintance.